recipes/linux-headers-generic/all/conanfile.py

```diff
@@ -0,0 +1,38 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.33.0"
+
+
+class LinuxHeadersGenericConan(ConanFile):
+    name = "linux-headers-generic"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.kernel.org/"
+    license = "GPL-2.0-only"
+    description = "Generic Linux kernel headers"
+    topics = ("linux", "headers", "generic")
+    settings = "os", "arch"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("linux-headers-generic supports only Linux")
+        if tools.cross_building(self):
```

Comment:
To properly detect cross-building, [we need a temporal workaround](https://github.com/conan-io/conan-center-index/pull/6622#discussion_r685814490):

```suggestion
        if hasattr(self, "settings_build") and tools.cross_building(self):
```